src/Packagist/WebBundle/Controller/UserController.php

```diff
@@ -270,6 +274,92 @@ public function deleteFavoriteAction(User $user, Package $package)
         return new Response('{"status": "success"}', 204);
     }
 
+    /**
+     * @Template()
+     * @Route("/users/{name}/2fa/", name="user_2fa_configure", methods={"GET"})
+     * @ParamConverter("user", options={"mapping": {"name": "username"}})
+     */
+    public function configureTwoFactorAuthAction(User $user)
+    {
+        if (!($this->isGranted('ROLE_DISABLE_2FA') || $user->getId() === $this->getUser()->getId())) {
+            throw new AccessDeniedException('You cannot change this user\'s two-factor authentication settings');
+        }
+
+        if ($backupCode = $this->get('session')->get('backup_code')) {
```

Comment:
the retrieval (and cleaning) of the backup code should be done only when looking at the page for yourselves IMO.

Reply:
Fixed and simplified to:

```php
if ($user->getId() === $this->getUser()->getId()) {
    $backupCode = $this->get('session')->remove('backup_code');
}
```

See commit https://github.com/composer/packagist/pull/1031/commits/0b4272099dc205665164eda71f4828482b5cc82fhttps://github.com/composer/packagist/pull/1031/commits/0b4272099dc205665164eda71f4828482b5cc82f